### ISSUE TITLE
MP3 GRP1 tag returned with leading null

### ISSUE THREAD

**Issue by darrell-k** [LMS-Community/Audio-Scan#7]
I worked around this with this code in Slim::Formats::MP3 - 

```
	my $s = Audio::Scan->scan( $file );

	if ( exists $s->{tags}->{GRP1} ) {
		$s->{tags}->{GRP1} =~ s/\x0//g; # scanner is returning leading null in GRP1 tag, which messes things up.
	}

```

**Comment by ralph-irving** [comment]
@darrell-k  Sorry I just noticed this issue yesterday.
Can you provide an mp3 file that exhibits the problem?
Is GRP1 tag mapped to the TIT1 for idv2.[34]?
Thanks.

**Comment by darrell-k** [comment]
@ralph-irving See https://forums.lyrion.org/forum/user-forums/logitech-media-server/1741878-multiple-album-versions?p=1742493#post1742493 for a summary of the GRP1/TIT1 confusion.

Here is the post which originally reported this issue: https://forums.lyrion.org/forum/user-forums/logitech-media-server/1737175-lyrion-music-server-9-0-0-classical-works-functionality-support?p=1758063#post1758063

I'll send you a forum PM with the test data.

**Comment by darrell-k** [comment]
There was a follow up on this, discussing different handling of UTF8/16 in ID3 v2.3 vs v2.4.  See https://forums.lyrion.org/forum/developer-forums/beta/1695133-announcement-classical-works-now-integrated-into-lms?p=1761190#post1761190 and the following few posts.

**Comment by ralph-irving** [comment]
Thanks for the files and references @darrell-k .

I now know why the GRP1 tag returns a leading "null" in audio-scan.
The GRP1 tag is unknown in the id3v2.x library included with audio-scan.  So the id3 support routines read the tag as binary data which returns the 03 encoding as the first byte.

```
000000E4  54 52 43 4B 00 00 00 03 00 00 03 30 TRCK.......0
000000F0  31 54 44 52 43 00 00 00 05 00 00 03 1TDRC.......
000000FC  31 39 38 32 47 52 50 31 00 00 00 0D 1982GRP1....
00000108  00 00 03 4C 61 73 74 20 45 6D 62 72 ...Last Embr
00000114  61 63 65 54 58 58 58 00 00 00 41 00 aceTXXX...A.
```
```
   TDRC, frame flags 0, size 5
    nfields 2: 0 5
    encoding: 3
    read utf8 string of 5 bytes: 1982
  GRP1, frame flags 0, size 13
    nfields 1: 14
    unknown returning raw data
```

As a test I added the GRP1 tag support and it's now parse correctly as utf8 data stripping the lead encoding byte.
```
  TDRC, frame flags 0, size 5
    nfields 2: 0 5
    encoding: 3
    read utf8 string of 5 bytes: 1982
  GRP1, frame flags 0, size 13
    nfields 2: 0 5
    encoding: 3
    read utf8 string of 13 bytes: Last Embrace
```

The GRP1 tag doesn't seem to be part of the spec so I'm unsure how best to handle it in audio-scan.  I can commit the change but am also look for suggestions.

I've been using this site for v2.4 tag references.  https://mutagen-specs.readthedocs.io/en/latest/id3/id3v2.4.0-frames.html

**Comment by ralph-irving** [comment]
There are other "experimental" ID3v2.x tags that have been added to Audio::Scan XSOP for example.
If LMS is using the GRP1 tag now then I see no reason I can't just go ahead and add it.

**Comment by darrell-k** [comment]
Makes sense to me, as the tagging software I've come across will read/write GRP1.

Just to aid my own understanding, is it correct that if a user adds a custom tag it ends up as TXXX:.... and is therefore treated as text, which explains why adding a custom tag of GROUPING would get around the problem?

**Comment by ralph-irving** [comment]
Yes, custom tags are stored as TXXX. 

For example, a custom tag named (User Frame) with the value (User Data) and are separated by a NULL.

Here's a hex dump of the tag described above;
```
00000168  2E 63 6F 6D 54 58 58 58 00 00 00 15 00 00 03 55 73 65 72 20 .comTXXX.......User
0000017C  46 72 61 6D 65 00 55 73 65 72 20 44 61 74 61 57 58 58 58 00 Frame.User DataWXXX.
```

and how Audio::Scan parses it.
```
  TXXX, frame flags 0, size 21
    nfields 3: 0 3 3
    encoding: 3
    read utf8 string of 11 bytes: User Frame
    read utf8 string of 10 bytes: User Data
```

Hope that helps.

I'll commit the GRP1 changes before releasing 1.11.

**Comment by ralph-irving** [comment]
I've created the release which includes support for GRP1 mp3 tags.

**Comment by darrell-k** [comment]
Many thanks.
